server/services/premisesService.test.ts

```diff
@@ -142,4 +143,94 @@ describe('PremisesService', () => {
       expect(premisesClient.find).toHaveBeenCalledWith(premises.id)
     })
   })
+
+  describe('getOvercapacityMessage', () => {
+    it('returns an empty string if not passed any dates', async () => {
+      premisesClient.capacity.mockResolvedValue([])
+
+      const result = await service.getOvercapacityMessage(token, premisesId)
+
+      expect(result).toBe('')
+    })
+    it('returns an empty string if passed dates that are not overcapacity', async () => {
+      premisesClient.capacity.mockResolvedValue([
+        dateCapacityFactory.build({ date: new Date(2023, 0, 1).toISOString(), availableBeds: 1 }),
+        dateCapacityFactory.build({ date: new Date(2023, 1, 1).toISOString(), availableBeds: 2 }),
+        dateCapacityFactory.build({ date: new Date(2023, 1, 2).toISOString(), availableBeds: 3 }),
+        dateCapacityFactory.build({ date: new Date(2023, 2, 2).toISOString(), availableBeds: 4 }),
+        dateCapacityFactory.build({ date: new Date(2023, 3, 2).toISOString(), availableBeds: 5 }),
+      ])
+
+      const result = await service.getOvercapacityMessage(token, premisesId)
+
+      expect(result).toBe('')
+    })
+
+    it('returns the correct string if passed a single date', async () => {
+      premisesClient.capacity.mockResolvedValue([
+        dateCapacityFactory.build({
+          date: new Date(2022, 0, 1).toISOString(),
+          availableBeds: -1,
+        }),
+      ])
```

Comment:
I think we should mock the resolved value of `getDateRangesWithNegativeBeds` here too, as that is what we're acting on. Maybe we mock the value of the `capacity` endpoint with stub, mock what we get from `getDateRangesWithNegativeBeds` and then check `getDateRangesWithNegativeBeds` was called with our `capacity` stub. 

Some psuedocode here:

```typescript

const capacityStub = [
  dateCapacityFactory.build(),
]

premisesClient.capacity.mockResolvedValue(capacityStub)

getDateRangesWithNegativeBeds.mockResolvedValue({start: new Date(2022, 0, 1).toISOString()})
```

Then that way we're clearer about what we're expecting back from that method, rather than relying on us passing stuff through the chain.

Does that make sense?

Reply:
done - although I opted to use the dates from the capacityStub in the return value from getDateRangesWithNegative beds as I think it highlights the link between the two